CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+# Core Developer Guide
+
+Welcome, new core developer!  The core team appreciate the quality of
+your work, and enjoy working with you; we have therefore invited you
+to join us.  Thank you for your numerous contributions to the project
+so far.
+
+You can see a list of all the current core developers on our
+[@zarr-developers/core-devs](https://github.com/orgs/zarr-developers/teams/core-devs)
+GitHub team. You should now be on that list too.
+
+This document offers guidelines for your new role.
+As a core team member, you gain the responsibility of shepherding
+other contributors through the review process; here are some
+guidelines for how to do that.
+
+## All Contributors Are Treated The Same
+
+As a core developer, you gain the ability to merge or approve
+other contributors' pull requests.  Much like nuclear launch keys, it
+is a shared power: you must merge *only after* another core has
+approved the pull request, *and* after you yourself have carefully
+reviewed it.  (See [Reviewing](#reviewing) and especially
```

Comment:
This sounds like we're saying pull requests require two approvals from core devs, is that right? But in the governance doc we say only one approval from a core dev is required?

Reply:
@alimanfoo : you mean in the case where a core dev has opened a PR `CORE_DEV_GUIDE` says that _another_ core dev must also approve? That's how I would read this paragraph as well. So options would include clarifying the `GOVERNANCE` section so that it's "one core dev" _other than yourself_ as well as making this section less strict so that core devs could merge their own PRs without further review.

Reply:
Maybe this illustrates the ambiguity, I actually read this the other way. I.e., I read this as saying, if a contributor opens a PR, then *two* core devs are required to approve. Which seemed inconsistent with the later section that said *one* core dev is required to approve.

Basically I'm very happy with the idea that at least *one* core dev is required to approve a PR, and if you're a core dev you can't approve your own PRs (which IIRC github won't let you do anyway?). Just wanting to remove any ambiguity about that.

Reply:
Pushed a change for "you may merge your own *only after*...". Does this suffice?

Reply:
Hi @joshmoore, that was good but seemed like content could be reorganised slightly for even more clarity. Pushed a commit, hope that's OK, happy to revise further.